addons/mrp/models/mrp_production.py

```diff
@@ -1786,6 +1786,10 @@ def _check_sn_uniqueness(self):
                         ('lot_id', '=', move_line.lot_id.id),
                         ('state', '=', 'done'),
                         ('location_dest_id.scrap_location', '=', True)
+                    ]) - self.env['stock.move.line'].search_count([
+                        ('lot_id', '=', move_line.lot_id.id),
+                        ('state', '=', 'done'),
+                        ('location_id.scrap_location', '=', True),
```

Comment:
Why 2 queries? Can't you include this leaf in the other domain?

Reply:
I don't think we can do it in a single query through the ORM as we want to get the difference between incoming and outgoing move in the scrap location. We could read all the move lines where either the location from or the location to is a scrap, and then do the difference in python, but I'm not sure it would be more efficient than using two `search_count`.